ts/WoltLabSuite/Core/Acp/Ui/Package/QuickInstallation.ts

```diff
@@ -0,0 +1,109 @@
+import { dboAction } from "../../../Ajax";
+import { isPlainObject } from "../../../Core";
+import * as Language from "../../../Language";
+import { innerError } from "../../../Dom/Util";
+import UiDialog from "../../../Ui/Dialog";
+
+let codeInput: HTMLInputElement;
+
+type InstallationCode = {
+  package: string;
+  password: string;
+  username: string;
+};
+
+type Response =
+  | {
+      queueID: number;
+      returnValues?: never;
+    }
+  | {
+      queueID?: never;
+      returnValues: {
+        template: string;
+      };
+    };
+
+function detectCode(): void {
+  const value = codeInput.value.trim();
+
+  let isValid = false;
+  if (value.length > 0) {
+    const decodedValue = window.atob(value);
+
+    let maybeJson: unknown;
+    try {
+      maybeJson = JSON.parse(decodedValue);
+    } catch {
+      // Skip invalid values.
+    }
+
+    if (isPlainObject(maybeJson)) {
+      const json = maybeJson as InstallationCode;
+      if (json.package && json.password && json.username) {
+        isValid = true;
+
+        void prepareInstallation(json);
+      }
+    }
+  }
+
+  if (isValid) {
+    innerError(codeInput, false);
+  } else {
+    innerError(codeInput, Language.get("wcf.acp.package.quickInstallation.code.error.invalid"));
+  }
+}
+
+async function prepareInstallation(data: InstallationCode): Promise<void> {
+  const response = (await dboAction("prepareInstallation", "wcf\\data\\package\\update\\PackageUpdateAction")
+    .payload({
+      packages: {
+        [data.package]: "",
+      },
+      authData: {
+        username: data.username,
+        password: data.password,
+        saveCredentials: false,
+      },
+    })
+    .dispatch()) as Response;
+
+  if (response.queueID) {
+    const installation = new window.WCF.ACP.Package.Installation(response.queueID, undefined, false);
+    installation.prepareInstallation();
+  } else if (response.returnValues) {
+    UiDialog.open(
+      {
+        _dialogSetup() {
+          return {
+            id: "foo",
+            source: null,
+          };
+        },
+      },
+      response.returnValues.template,
+    );
```

Comment:
This still is stubbed out. Please add proper error handling.

Reply:
This was intentionally kept this way. We cannot recover from this and I rather have a proper error message than hiding it behind meaningless inline errors.

Reply:
At the very least you should give this dialog a proper title; and also a proper `id` instead of `"foo"` :grin: